contracts/Router.sol

```diff
@@ -0,0 +1,261 @@
+        // cannot overflow.
+        _erc20TotalReleased[token] += payment;
+        unchecked {
+            _erc20Released[token][account] += payment;
+        }
+
+        SafeERC20.safeTransfer(token, account, payment);
+        emit ERC20PaymentReleased(token, account, payment);
+    }
+
+    /**
+     * @dev internal logic for computing the pending payment of an `account` given the token historical balances and
+     * already released amounts.
+     */
+    function _pendingPayment(
+        address account,
+        uint256 totalReceived,
+        uint256 alreadyReleased
+    ) private view returns (uint256) {
+        return (totalReceived * _shares[account]) / _totalShares - alreadyReleased;
+    }
+
+    /**
+     * @dev Add a new payee to the contract.
+     * @param account The address of the payee to add.
+     * @param shares_ The number of shares owned by the payee.
+     */
+    function _addPayee(address account, uint256 shares_) private {
+        require(account != address(0), "PaymentSplitter: account is the zero address");
+        require(shares_ > 0, "PaymentSplitter: shares are 0");
+        require(_shares[account] == 0, "PaymentSplitter: account already has shares");
+
+        _payees.push(account);
+        _shares[account] = shares_;
+        _totalShares = _totalShares + shares_;
+        emit PayeeAdded(account, shares_);
+    }
+
+    function _removePayee(address account) private {
+        require(account != address(0), "PaymentSplitter: account is the zero address");
+        require(_shares[account] > 0, "PaymentSplitter: account has no shares");
+
+        uint256 shares_ = _shares[account];
+        _shares[account] = 0;
+        _totalShares = _totalShares - shares_;
+
+        for (uint256 i = 0; i < _payees.length; i++) {
+            if (_payees[i] == account) {
+                _payees[i] = _payees[_payees.length - 1];
+                _payees.pop();
+                break;
+            }
+        }
+
+        emit PayeeRemoved(account, shares_);
+    }
+
+    function _adjustShare(address account, uint256 shares_) private {
+        require(account != address(0), "PaymentSplitter: account is the zero address");
+        require(shares_ > 0, "PaymentSplitter: shares are 0");
+        require(_shares[account] > 0, "PaymentSplitter: account has no shares");
+
+        uint256 oldShares = _shares[account];
+        _shares[account] = shares_;
+        _totalShares = _totalShares - oldShares + shares_;
+
+        emit PayeeShareAdjusted(account, shares_, oldShares);
+    }
+
+    function addPayee(address account, uint256 shares_) external onlyOwner {
+        _addPayee(account, shares_);
+    }
+
+    function removePayee(address account) external onlyOwner {
+        _removePayee(account);
+    }
+
+    function adjustShare(address account, uint256 shares_) external onlyOwner {
+        _adjustShare(account, shares_);
+    }
```

Comment:
What's the point of these two functions? We can just add the `external onlyOwner` modifiers to the functions above, `_removePayee` and `_adjustShare` are only used here